sips/sip-x/sip-x-improved-stacking.md

```diff
@@ -0,0 +1,206 @@
+tokens are locked for a locking period that always ends at the beginning of a cycle, i.e. after the
+prepare phase. Therefore, the current implementation of Stacking includes a period where Stacks
+tokens are unlocked and ineligible for stacking rewards (Cooldown Cycle).
+
+The current PoX contract allows users to lock Stacks tokens during the prepare phase. This is too late for the next cycle
+and the tokens are locked without earning stacking rewards for one cycle.
+
+The cooldown cycle for unstacking from a pool presents a problem for network decentralization.
+As a Stacker, when I delegate to a signer and that signer does not perform and gets low yield,
+I get penalized for switching. In today's model with cooldown cycles, users get double penalized
+if a signer does not perform. Even if there are penalties for signers not performing / being down,
+the switching costs are too high for users to switch (two weeks' worth of yield).
+
+Furthermore, users cannot change their
+stacking settings (decrease amount, change PoX reward address, etc.) without an unlocked cycle during which users cannot earn
+stacking rewards.
+
+## Proposed Solution
+
+This SIP proposes a new Proof of Transfer (PoX) contract without
+the flow for solo stacking and moves responsibilities from pool operators to signers who were introduced in the Nakamoto upgrade. It also
+defines a new locking mechanism that allows users to switch from one signer to another with a single contract call and without a cooldown period.
+
+# Specification
+
+Applying these upgrades to the Stacks blockchain requires a consensus-breaking network upgrade,
+in this case, a hard fork. Like other such changes, this will require a new Stacks epoch.
+In this SIP, we will refer to this new epoch as Stacks 3.2.
+
+## Delegated Stacking only
+
+The following PoX contract functions shall be removed:
+
+- stack-stx
+- stack-increase
+- stack-extend
+
+## Locking Post Conditions
+
+The following type of post conditions shall be added to the current definition in SIP-005 (`TransactionPostCondition`)
+
+- `LockingLimit(PostConditionPrincipal, FungibleConditionCode, u64)`
+
+A transaction (using Deny mode) with this post condition will abort if the locked Stacks tokens of the principal do not satisfy the provided conditions. The logic for the conditions follows that of STX transfer.
+
+## Automatic Extend and Locking Period
+
+The following function shall be removed:
+
+- `delegate-stack-extend`
+
+In addition, the stacking settings for each Stacker (user with delegation) shall be applied for the next stacking cycle if the user did not signal the end of stacking by calling `revoke-delegate-stx` 200 blocks before the start of the next cycle.
+
+This results in the following structure of the stacking cycle:
+
+- bitcoin block 1-1900: stacking as usual, user can signal change of stacking settings.
+- bitcoin block 1901-2000: stacking as usual, signers can still aggregate stacking changes.
```

Comment:
If a user tries to signal a change in their stacking settings, are is it possible for it to apply to the next cycle? Whatever the answer is, can we add it to this SIP?